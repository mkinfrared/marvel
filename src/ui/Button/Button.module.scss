@import "styles/colors";

.Button {
  background-color: $primary-main;
  border: none;
  border-radius: 4px;
  box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2);
  cursor: pointer;
  outline: none;
  padding: 6px 16px;
  text-transform: uppercase;
  transition: filter 250ms linear;

  &:hover {
    filter: brightness(0.8);
  }

  &:active {
    filter: brightness(1.1);
  }

  &:disabled {
    background-color: rgba(33, 33, 33, 0.69);
    color: rgba(255, 255, 255, 0.41);
    cursor: auto;
  }
}
