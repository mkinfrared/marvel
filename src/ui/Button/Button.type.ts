export type ButtonProps = React.ComponentProps<"button">;
