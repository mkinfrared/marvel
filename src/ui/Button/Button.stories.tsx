import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import { Button } from "./Button";
import { ButtonProps } from "./Button.type";

export default {
  title: "UI/Button",
  component: Button
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args}>Hello</Button>;
const Primary = Template.bind({});

export { Primary };
