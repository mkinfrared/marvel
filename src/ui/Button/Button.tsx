import React from "react";

import css from "./Button.module.scss";
import { ButtonProps } from "./Button.type";

const Button = ({
  children,
  onClick,
  disabled = false,
  type = "button"
}: ButtonProps) => {
  return (
    <button
      className={css.Button}
      onClick={onClick}
      type={type}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

export { Button };

export default React.memo(Button);
