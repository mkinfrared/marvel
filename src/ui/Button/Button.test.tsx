import { render } from "@testing-library/react";
import React from "react";

import { Button } from "./Button";

describe("<Button />", () => {
  it("should be defined", () => {
    const { container } = render(<Button />);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
});
