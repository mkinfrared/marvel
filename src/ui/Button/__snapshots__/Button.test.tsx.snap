// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> match the snapshot 1`] = `
<div>
  <button
    class="Button"
    type="button"
  />
</div>
`;
